pkg/trafficrouting/manager.go

```diff
@@ -447,6 +451,15 @@ func (m *Manager) createCanaryService(c *TrafficRoutingContext, cService string,
 	for i := range canaryService.Spec.Ports {
 		canaryService.Spec.Ports[i].NodePort = 0
 	}
+	if len(c.CanaryServiceSelectorPatch) > 0 {
```

Comment:
len comparison is not necessary,  we can iterator over an nil map

Reply:
Fixed